src/services/role-assignment-service.ts

```diff
@@ -1,13 +1,18 @@
 /* eslint-disable no-await-in-loop */
-import { Client, FetchMemberOptions } from 'discord.js';
+import { Client } from 'discord.js';
+import { RoleUpdateData } from '../models/role-update-data';
 
 export class RoleAssignmentService {
-  static async updateRoles(data: RoleUpdateData, client: Client): Promise<boolean> {
+  static async updateRoles(data: RoleUpdateData, client: Client): Promise<void> {
     const guild = await client.guilds.fetch(data.guildID).catch((err) => console.log(err));
-    if (!guild) { return false; }
+    if (!guild) { 
+      throw new Error(`could get guild: ${data.guildID}`);
+    }
 
     const member = await guild.members.fetch({ 'user': data.userID, 'cache': true, 'force': true }).catch((err) => console.log(err))/*cache.get(data.userID)*/;
```

Comment:
.fetch does not return undefined/null
no need to check for it if you remove the unnecessary .catch